### PR TITLE
Switch Windows CI jobs to G5 runners

### PR DESCRIPTION
**Pull request by huydhn**
### Changelist

* Change Windows TORCH_CUDA_ARCH_LIST from `7.0` to `8.6` to compatible with NVIDIA A10G TPU
* Correctly disable some tests that requires flash attention, which is not available on Windows at the moment. This has been fixed by https://github.com/pytorch/pytorch/pull/91979
* G5 runner has `AMD EPYC 7R32` CPU, not an Intel one
  * This seems to change the behavior of `GetDefaultMobileCPUAllocator` in `cpu_profiling_allocator_test`.  This might need to be investigated further (TODO: TRACKING ISSUE).  In the meantime, the test has been updated accordingly to use `GetDefaultCPUAllocator` correctly instead of `GetDefaultMobileCPUAllocator` for mobile build
  * Also one periodic test `test_cpu_gpu_parity_nn_Conv3d_cuda_float32` fails with Tensor not close error when comparing grad tensors between CPU and GPU. This is fixed by turning off TF32 for the test.

###  Performance gain

* (CURRENT) p3.2xlarge - https://hud.pytorch.org/tts shows each Windows CUDA shards (1-5 + functorch) takes about 2 hours to finish (duration)
* (NEW RUNNER) g5.4xlarge - The very rough estimation of the duration is 1h30m for each shard, meaning a half an hour gain (**25%**)

### Pricing

On demand hourly rate:

* (CURRENT) p3.2xlarge: $3.428. Total = Total hours spent on Windows CUDA tests * 3.428
* (NEW RUNNER) g5.4xlarge: $2.36. Total = Total hours spent on Windows CUDA tests * Duration gain (0.75) * 2.36

So the current runner is not only more expensive but is also slower.  Switching to G5 runners for Windows should cut down the cost by (3.428 - 0.75 * 2.36) / 3.428 = **~45%**

### Rolling out

https://github.com/pytorch/test-infra/pull/1376 needs to be reviewed and approved to ensure the capacity of the runner before PR can be merged.

### CODE REVIEW

File: .github/workflows/_win-test.yml [hunk 1]
```diff
@@ -144,7 +144,7 @@ jobs:
           NUM_TEST_SHARDS: ${{ matrix.num_shards }}
           TEST_CONFIG: ${{ matrix.config }}
           PR_BODY: ${{ github.event.pull_request.body }}
-          TORCH_CUDA_ARCH_LIST: "7.0"
+          TORCH_CUDA_ARCH_LIST: "8.6"
```
Comment (ZainRizvi) [inline .github/workflows/_win-test.yml:147]: Looking at https://github.com/search?q=repo%3Apytorch%2Fpytorch%20TORCH_CUDA_ARCH_LIST&type=code, there are a couple places where we default this value to 5.2

Would it make sense to change those defaults as well?

**Review by ZainRizvi — APPROVED** [review]

File: .github/workflows/_win-test.yml (hunk 1 above)
Comment (huydhn) [reply .github/workflows/_win-test.yml:147]: Yeah, I can change that.  `5.2` is probably a legacy default that is still part of the code till date.

**Review by malfet — APPROVED** [review]

File: .github/workflows/_win-test.yml (hunk 1 above)
Comment (seemethere) [reply .github/workflows/_win-test.yml:147]: 5.2 is incredibly old wouldn't recommend doing that

File: .github/workflows/_win-test.yml [hunk 2]
```diff
@@ -63,6 +63,21 @@ jobs:
         run: |
           git config --global core.symlinks true
 
+      - name: Setup SSH (Click me for login details)
+        uses: pytorch/test-infra/.github/actions/setup-ssh@main
+        with:
+          github-secret: ${{ secrets.GITHUB_TOKEN }}
+          instructions: |
```
Comment (seemethere) [inline .github/workflows/_win-test.yml:70]: #til that we added support for this :D

**Review by seemethere — APPROVED** [review]

File: .github/workflows/_win-test.yml (hunk 1 above)
Comment (huydhn) [reply .github/workflows/_win-test.yml:147]: Oh, I mean removing 5.2 as the default as it's indeed very old :)

File: .github/workflows/_win-test.yml (hunk 2 above)
Comment (malfet) [reply .github/workflows/_win-test.yml:70]: Was about to comment, that you've reviewed the change that enabled it, but then check and indeed you didn't :P

File: test/test_modules.py [hunk 3]
```diff
@@ -488,6 +489,7 @@ def test_gradgrad(self, device, dtype, module_info, training):
         self._test_gradients_helper(device, dtype, module_info, training, gradgradcheck)
 
     @onlyCUDA
+    @with_tf32_off  # Turn off TF32 to compute at full precision https://github.com/pytorch/pytorch/issues/86798
```
Comment (malfet) [inline test/test_modules.py:492]: 👍 +1

**Pull request closed without merging**